### ISSUE TITLE
Steamshovel requires Qt >= 4.8 (Trac #478)

### ISSUE THREAD

**Issue by icecube-trac** [icecube-trac/tix2#477]
Steamshovel will build successfully with the current Qt version in I3_PORTS (4.6.4).  However, at least one significant feature will not work correctly under this version of Qt.  

Under Qt 4.6, when steamshovel is running in in-shell console mode (i.e. there is a python prompt on the shell that spawned the steamshovel binary; this behavior can be forced with `steamshovel --console`), there is no way to access the scripting object 'window.gl.scenario'.  Attempts to read this value produce this error message:

    QMetaMethod::invoke: Unable to invoke methods with return values in queued connections
    ERROR (qmeta): Queued property read failed (qmeta_args.cpp:102 in T scripting::qmeta_detail::property_read(QMetaProperty, QObject*) [with T = Scenario*])

This issue does not prevent basic usage of steamshovel, but will affect users who are trying to use the steamshovel scripting layer for automation tasks.  Building steamshovel with Qt >= 4.8 fixes the issue.  

The underlying issue is the subject of this Qt bug:

https://bugreports.qt-project.org/browse/QTBUG-10440

Migrated from https://code.icecube.wisc.edu/ticket/478
```json
{
    "status": "closed", 
    "changetime": "2019-01-11T23:06:32", 
    "description": "Steamshovel will build successfully with the current Qt version in I3_PORTS (4.6.4).  However, at least one significant feature will not work correctly under this version of Qt.  \n\nUnder Qt 4.6, when steamshovel is running in in-shell console mode (i.e. there is a python prompt on the shell that spawned the steamshovel binary; this behavior can be forced with `steamshovel --console`), there is no way to access the scripting object 'window.gl.scenario'.  Attempts to read this value produce this error message:\n\n    QMetaMethod::invoke: Unable to invoke methods with return values in queued connections\n    ERROR (qmeta): Queued property read failed (qmeta_args.cpp:102 in T scripting::qmeta_detail::property_read(QMetaProperty, QObject*) [with T = Scenario*])\n\nThis issue does not prevent basic usage of steamshovel, but will affect users who are trying to use the steamshovel scripting layer for automation tasks.  Building steamshovel with Qt >= 4.8 fixes the issue.  \n\nThe underlying issue is the subject of this Qt bug:\n\nhttps://bugreports.qt-project.org/browse/QTBUG-10440\n\n", 
    "reporter": "sjackso", 
    "cc": "nwhitehorn", 
    "resolution": "fixed", 
    "_ts": "1547247992066444", 
    "component": "steamshovel", 
    "summary": "Steamshovel requires Qt >= 4.8", 
    "priority": "normal", 
    "keywords": "", 
    "time": "2013-12-30T17:57:13", 
    "milestone": "", 
    "owner": "nega", 
    "type": "defect"
}
```

**Comment by icecube-trac** [comment]
Trac update at 20141003T19:24:39: 
* nega commented:

> we're not providing Qt anymore. no update to ports.
> 
> fixed in [1942/IceTray]


* nega changed status from "new" to "closed"
* nega changed resolution from "" to "fixed"

**Comment by icecube-trac** [comment]
Trac update at 20190111T23:06:32: nwhitehorn commented:

> In changeset 169865:
> 
> ```text
> #!CommitTicketReference repository="IceCube" revision="169865"
> make 4.8 the minimum version of qt
> 
> closes #477
> ```
